### ISSUE TITLE
Alignment on a specific element of the stack

### ISSUE THREAD

**Issue by lucabartoletti** [linkedin/LayoutKit#20]
Let's say a stack contains two layouts [A][B], if I set .center as aligment for the stack, than the stack's container will set the stack position based on the mid point of it's size. There are cases where I want to center the stack on a specifc element. 

For example in this Stack I have to elements the dot and the text. That text should be centered.
![Sample](http://i.imgur.com/OVvFOff.png)
A solution to this specific example could be to create an empty SizeLayout as third element of the stack with the same width than the first one. But this could be more complicated with dynamic elements and it should be possible without have to create more layout elements than necessary.

**Comment by nicksnyder** [comment]
I agree that this is a useful feature to have in general. One idea I have would be to add a `StackLayout` distribution of `case centerSublayout(index: Int)` which centers on the sublayout at that index. Thoughts?

In the meantime, you have a few options:
- Could use `InsetLayout` to pad the right side of the stack by the amount of the orange circle
- Write a custom Layout that owns or inherits a StackLayout and makes the appropriate adjustments in `measure` and `position`
- Create a pull request and propose something :)
